htdocs/main.inc.php

```diff
@@ -461,15 +461,16 @@ function analyseVarsForSqlAndScriptsInjection(&$var, $type)
 if (!defined('NOLOGIN') && !defined('NOIPCHECK') && !empty($dolibarr_main_restrict_ip)) {
 	$listofip = explode(',', $dolibarr_main_restrict_ip);
 	$found = false;
+	$user_ip = getUserRemoteIP();
 	foreach ($listofip as $ip) {
 		$ip = trim($ip);
-		if ($ip == $_SERVER['REMOTE_ADDR']) {
+		if ($ip == $user_ip) {
```

Comment:
This break the security:
getUserRemoteIP return a non reliable IP for security (try to guess the end user IP but can be spoofed by another one by a hacker just with a simple http header). 
If we restrict acces on some IP, test must no be done on a value forged by the end user or a intermediate server but on the REAL IP that connect to the server. So it must be $_SERVER['REMOTE_ADDR'] (the only value that is the real ip viewed by the server).
If you need to connect using a proxy, you must allow the IP of the proxy. This is the goal of the protection to not allow a proxy if ip is another one.

Reply:
Ok for the explication, so I reverted that.
However, with regards to security, when IP verification is done, I've seen it mostly done for the entire proxy chain.  So maybe there is an improvement to do on this check.